apps/full-site-editing/full-site-editing-plugin/global-styles/src/global-styles-sidebar.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * External dependencies
+ */
+import { PluginSidebar } from '@wordpress/edit-post';
+import { Button, PanelBody } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { PluginSidebarMoreMenuItem } from '@wordpress/edit-post';
+
+/**
+ * Internal dependencies
+ */
+import FontPairingsPanel from './font-pairings-panel';
+import FontSelectionPanel from './font-selection-panel';
+import { GlobalStylesIcon } from './icon';
+import { FONT_BASE, FONT_HEADINGS } from './constants';
+
+const ANY_PROPERTY = 'ANY_PROPERTY';
+
+const isFor = filterProperty => option =>
+	option.prop === ANY_PROPERTY || option.prop === filterProperty;
+
+const toOption = font => {
+	if ( typeof font === 'object' ) {
+		const { label, value, prop = ANY_PROPERTY } = font;
+		return { label, value, prop };
+	}
+	return { label: font, value: font, prop: ANY_PROPERTY };
+};
+const isNotNull = option => option.value !== null && option.label !== null;
+
+const toOptions = ( options, filterProperty ) =>
+	! options
+		? []
+		: options
+				.map( toOption )
+				.filter( isNotNull )
+				.filter( isFor( filterProperty ) );
+
+const PanelActionButtons = ( { hasLocalChanges, resetAction, publishAction, className } ) => (
+	<div className={ className }>
+		<Button disabled={ ! hasLocalChanges } isDefault onClick={ resetAction }>
+			{ __( 'Reset' ) }
+		</Button>
+		<Button
+			disabled={ ! hasLocalChanges }
+			style={ { marginLeft: '1em' } }
+			isPrimary
+			onClick={ publishAction }
+		>
+			{ __( 'Publish' ) }
+		</Button>
+	</div>
+);
+
+export default ( {
+	fontHeadings,
+	fontHeadingsDefault,
+	fontBase,
+	fontBaseDefault,
+	fontPairings,
+	fontOptions,
+	siteName,
+	publishOptions,
+	updateOptions,
+	hasLocalChanges,
+	resetLocalChanges,
+} ) => {
+	const publish = () =>
+		publishOptions( {
+			[ FONT_BASE ]: fontBase,
+			[ FONT_HEADINGS ]: fontHeadings,
+		} );
+	return (
+		<>
+			<PluginSidebarMoreMenuItem icon={ <GlobalStylesIcon /> } target="global-styles">
+				{ __( 'Global Styles' ) }
+			</PluginSidebarMoreMenuItem>
+			<PluginSidebar
+				icon={ <GlobalStylesIcon /> }
+				name={ 'global-styles' }
+				title={ __( 'Global Styles' ) }
+				className="global-styles-sidebar"
+			>
+				<PanelBody>
+					<p>
+						{ __( 'You are customizing ' ) }
+						<strong>{ siteName }</strong>.
+					</p>
+					<p>{ __( 'Any change you make here will apply to the entire website.' ) }</p>
+					{ hasLocalChanges ? (
+						<div>
+							<p>
+								<em>{ __( 'You have unsaved changes.' ) }</em>
+							</p>
+							<PanelActionButtons
+								hasLocalChanges={ hasLocalChanges }
+								publishAction={ publish }
+								resetAction={ resetLocalChanges }
+							/>
+						</div>
+					) : null }
```

Comment:
instead of `{ has ? (<div/>) : null }` you can also do `{ has && (<div/>) }`

Reply:
I'm fine with the existing choice if there isn't any reason people find it concerning.

Reply:
`{ condition && <Element /> }` is the pattern we use elsewhere in Calypso. 👍